API_request_test.py

```diff
@@ -12,16 +46,24 @@
 # https://www.weatherapi.com/signup.aspx
 # The url is set to get forecast data. For historical data it should be changed
 
+# KeyError exception handling
 url_wheatherapi = 'https://api.weatherapi.com/v1/forecast.json'
-API_KEY_wheatherapi = <paste here your personal API key>
+try:
+    API_KEY_wheatherapi = keys_dict['wheatherapi.com']
+except KeyError:
+    print("The key name in the file better_weather_keys.txt is incorrect, should be 'wheatherapi.com'")
 
 # You can change the name of the city <q parameter>, search engine is quite efficient
 # days=1 means that you get hourly forecast for one date, the day you are make a request. 
 # To get forecast for next day you need additional parameter
-params_wheatherapi = dict(key=API_KEY_wheatherapi, q='Przemysl', days=1) 
+params_wheatherapi = dict(key=API_KEY_wheatherapi, q=args.location, days=1) 
 
 resonse_wheatherapi = requests.get(url_wheatherapi, params=params_wheatherapi)
 
+# check if lack of proper request response is due to 403 error
+if resonse_wheatherapi.status_code == 403:
+    print("Supplied API key for wheatherapi.com is wrong.")
```

Comment:
Actually, you're checking the response code here only (which makes sense btw), not if the response is proper/lacking anything. Yes, I'm nitpicking, but it doesn't reflect what's going on here.

Also, `403` is not the only option you may bump into, check out the possible [400's](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) and [500's](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses). It's a lot of things that may go wrong, and you may want to implement handling it in the next iterations, but for now, I'd consider handling at least the `500` code as well. Another way would be to check if the response code falls under any error (number in range) and print the actual error returned by the server (a properly designed API should return such information apart from the code). I'm ok with both options. Also, if there is an error, is the print the only thing to do in this place (should we continue the code execution)?

Reply:
As we discussed, I agree that I should handle here more of the potential errors/responses, but for now I would leave it as it is. I would work with this when writing the appropriate component. 
However, I added exit() function after the print(), because we should not continue the code execution after the error occurs.